res/controllers/Hercules-DJ-Console-RMX-2-scripts.js

```diff
@@ -98,62 +833,574 @@ DJCRMX2.Deck.prototype.wheelTurn = function (value) {
- */
-DJCRMX2.wheelPress = function (channel, control, value, status, group) {
-    var deck = 0;     
-    if (group == "[Channel1]") {
-        deck = 1;
-    } else  if (group == "[Channel2]") {
-        deck = 2;
+
+    if (this.isShifted) {
+        newValue = newValue * DJCRMX2.jogwheelShiftMultiplier;
+    }
+
+    if (engine.isScratching(deck)) {
+        engine.scratchTick(deck, newValue);
     } else {
-        return; 
+        engine.setValue(this.currentDeck, 'jog', newValue * DJCRMX2.jogwheelSensivity);
     }
+};
+
+DJCRMX2.wheelPress = function(channel, control, value, status, group) {
+    var deck = group.match(/\d+/)[0];
     DJCRMX2.decks[deck].wheelPress(value);
-}
-
- 
-/* [ Function wheelTurn ]
- * Pays attention to the current deck, checks scratching, affects the
- * song accordingly.
- */
-DJCRMX2.wheelTurn = function (channel, control, value, status, group) {
-    var deck = 0;     
-    if (group == "[Channel1]") {
-        deck = 1;
-    } else  if (group == "[Channel2]") {
-        deck = 2;
+};
+
+DJCRMX2.wheelTurn = function(channel, control, value, status, group) {
+    var deck = group.match(/\d+/)[0];
+    DJCRMX2.decks[deck].wheelTurn(value);
+};
+
+DJCRMX2.Master = function() {
+    var ma = this;
+    ma.group = '[Master]';
+    this.group = ma.group;
+
+    this.shiftButton = function(channel, control, value, status, group) {
+        if (value === 0x7F) {
+            this.shift();
+        } else {
+            this.unshift();
+        }
+    };
+
+    this.masterVolumeKnob = new components.Pot({
+        inKey: 'gain',
+    });
+
+    this.crossfader = new components.Pot({
+        inKey: 'crossfader',
+    });
+
+    this.headMixKnob = new components.Pot({
+        inKey: 'headMix',
+    });
+
+    this.scratchButton = new components.Button({
+        midi: [0x90, DJCRMX2.generalLeds.scratch],
+        type: components.Button.prototype.types.toggle,
+        unshift: function() {
+            this.inKey = 'headSplit';
+            this.outKey = 'headSplit';
+            this.disconnect();
+            this.connect();
+            this.trigger();
+        },
+        shift: function() {
+            this.inKey = 'maximize_library';
+            this.outKey = 'maximize_library';
+            this.disconnect();
+            this.connect();
+            this.trigger();
+        },
+    });
+
+    if (DJCRMX2.vuMeterOutputMaster) {
+        engine.connectControl(this.group, 'VuMeterL', 'DJCRMX2.vuMeterLeds');
+        engine.connectControl(this.group, 'VuMeterR', 'DJCRMX2.vuMeterLeds');
+    }
+
+    this.reconnectComponents(function(c) {
+        if (c.group === undefined) {
+            c.group = ma.group;
+        }
+    });
+};
+DJCRMX2.Master.prototype = new components.ComponentContainer();
+
+DJCRMX2.Library = function() {
+    var lib = this;
+    lib.group = '[Library]';
+    this.group = lib.group;
+
+    this.shiftButton = function(channel, control, value, status, group) {
```

Comment:
Aren't the shift buttons in the decks?

Reply:
With the shift buttons I had a problem because only some buttons have shift functionality and some not. But I wanted to have a shifted functionality on some buttons without it's own shifted MIDI message. That's why the shift button is declared in each group and there's also a call (don't really know how to spell it) for each group in the xml file which is triggered from the same shift button MIDI message. I thought this would be the best way to do it...if you have a nicer idea, just let me know.